astropy/stats/lombscargle/implementations/main.py

```diff
@@ -0,0 +1,322 @@
+
+
+def available_methods():
+    methods = ['auto', 'slow', 'chi2', 'cython']
+
+    # Numpy 1.8 or newer required for fast algorithms
+    if hasattr(np.ufunc, 'at'):
+        methods.extend(['fast', 'fastchi2'])
+
+    try:
+        import scipy
+        methods.append('scipy')
+    except ImportError:
+        pass
+    return methods
+
+
+def validate_inputs(t, y, dy=None, frequency=None, strip_units=True):
+    """Validation of input shapes & units
+
+    This utility function serves a few purposes:
+
+    - it validates that the shapes of t, y, and dy match, and broadcasts
+      them to a common 1D shape
+    - if any of t, y, day, or frequency are astropy Quantities (i.e. have
+      units attached), it validates that the units are compatible, and does
+      any necessary unit conversions
+    - if ``strip_units == True``, it strips units from all the arrays
+      before returning them.
+    - all relevant units are returned in ``unit_dict``
+
+    Parameters
+    ----------
+    t, y : array_like or Quantity
+    dy, frequency : array_like or Quantity (optional)
+    strip_units : bool (optional, default=True)
+        if True, the returned quantities will have units stripped.
+
+    Returns
+    -------
+    t, y, dy, frequency : ndarray, Quantity, or None
+        reshaped and/or unit-stripped arrays
+    unit_dict : dict
+        dictionary of relevant units
+    """
+    if dy is None:
+        t, y = broadcast_arrays(t, y, subok=True)
+    else:
+        t, y, dy = broadcast_arrays(t, y, dy, subok=True)
+
+    if t.ndim != 1:
+        raise ValueError("Input times & data must be one-dimensional")
+
+    has_units = any(isinstance(arr, units.Quantity)
+                    for arr in (t, y, dy, frequency))
+
+    if has_units:
+        power_unit = units.dimensionless_unscaled
+
+        t = units.Quantity(t)
+        y = units.Quantity(y)
+
+        if frequency is not None:
+            frequency = units.Quantity(frequency)
+            if not t.unit.is_equivalent(1. / frequency.unit):
+                raise ValueError("Units of frequency not equivalent to "
+                                 "units of 1/t")
+            t = units.Quantity(t, unit=1. / frequency.unit)
+
+        if dy is not None:
+            dy = units.Quantity(dy)
+            if not y.unit.is_equivalent(dy.unit):
+                raise ValueError("Units of y not equivalent to units of dy")
+            dy = units.Quantity(dy, unit=y.unit)
+    else:
+        power_unit = 1
+
+        t = np.asarray(t)
+        y = np.asarray(y)
+        if dy is not None:
+            dy = np.asarray(dy)
+
+    def get_unit(val):
+        if isinstance(val, units.Quantity):
+            return val.unit
+        else:
+            return 1
+
+    unit_dict = {'t': get_unit(t),
+                 'y': get_unit(y),
+                 'dy': get_unit(y),
+                 'frequency': 1. / get_unit(t),
+                 'power': power_unit}
+
+    def unit_strip(arr):
```

Comment:
In the case of a scalar, your version returns a scalar value rather than a 0-dimensional array and this breaks things down the line.